cdc/processor/sinkmanager/redo_cache.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sinkmanager
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/sorter"
+)
+
+// redoEventCache caches events fetched from EventSortEngine.
+type redoEventCache struct {
+	capacity  uint64
+	allocated uint64
+
+	mu     sync.Mutex
+	tables map[model.TableID]*eventsAndSize
+}
+
+func newRedoEventCache(capacity uint64) *redoEventCache {
+	return &redoEventCache{
+		capacity:  capacity,
+		allocated: 0,
+		tables:    make(map[model.TableID]*eventsAndSize),
+	}
+}
+
+func (r *redoEventCache) getAppender(tableID model.TableID) *eventsAndSize {
```

Comment:
This method's name looks a little bit strange to me. Because eventsAndSize does not look like an appender more like a cache item.